guides-markdown/wallets.md

```diff
@@ -0,0 +1,403 @@
+	console.log(jdExtendedPrivateKey.xprivkey());
+	//tprv8gryhi3CLWEBGRvWMSL7w1YyY3bHr4w3XBjt75vm842mntgv9Tvvzsc8Bf3NZt13ydD5QZaJVShMudE33egMhSLnEM41t5UUhRj5wA5u8Sc
+
+	//for good measure, let's get the private key for John's first receiving address
+	const branch = 0; //the branch for receiving addresses
+	const index = 0; //index of 0 means the first key among the receiving addresses
+	const jdPrivateKey0 = jdExtendedPrivateKey.derive(branch).derive(index);
+	const jdWalletKey0 = WalletKey.fromHD(jdAccount, jdPrivateKey0, branch, index);
+	//the private key below can be imported into almost any bitcoin wallet, HD or non-HD, and used to spend coins from the corresponding address
+	console.log(jdWalletKey0.getPrivateKey('base58')); 
+	//cNZfR3NhQ9oCP3pTjvPZETUuTWZo2k6EXtfczvbWyv7FdjMhppvJ
+```
+
+#### Generating Mnemonics and Recovering Keys
+
+Finally, you can create mnemonics manually and seed new wallets with them. And if you need to generate keys from a mnemonic you provide - either by recreating a wallet or by extracting specific keys - you can do that as well.
+
+```javascript
+	//can we generate a mnemonic with twice as many bits of entropy to future proof against brute force attacks from the next millenium? sure we can.
+	const mnemonic24 = new Mnemonic({bits: 256});
+	console.log(mnemonic24.toString());
+	//page unknown ladder thunder airport merry run ball inject clinic danger valley equip consider normal twist casual duck essay almost trade regular two segment
+
+	//what if we need to recover the wallet we've created above? no problem.
+	const mnemonic = new Mnemonic('again choose noble warrior print thrive post glare movie glove animal legal');
+	const masterKey = HD.fromMnemonic(mnemonic);
+	//this wallet will generate all the same accounts, keys, addresses, and tokens if swapped in for the 'wallet' variable in examples above
+	const recoveredWallet = await wdb.create({master: masterKey});
+	
+	//we can also recover only the keys for John Doe's first receiving address without recreating the wallet or account
+	//this time we'll skip instantiating the account and use the BIP44 path for the second account, first branch, first index
+	const jdRecoveredPrivateKey = masterKey.derivePath("m/44'/1'/1'/0/0");
+	const jdKeyRing = new KeyRing(jdRecoveredPrivateKey); 
+	//our output should be the same as what we logged to the console earlier in this example
+	console.log(jdKeyRing.getAddress('string'));
+	//muCSbWC6z1tAr2i1M5BKPWEZ8zapzcKfKh
+	console.log(jdKeyRing.getPrivateKey('base58'));
+	//cNZfR3NhQ9oCP3pTjvPZETUuTWZo2k6EXtfczvbWyv7FdjMhppvJ
+}
+```
+
+### Command Line Examples against a Local bcoin Server
+
+If you already have bcoin [set up as a full node](https://github.com/bcoin-org/bcoin/wiki/Beginner's-Guide), you can use the command line to create wallets and demonstrate some of the topics I discussed in this guide. My examples below are against a running testnet bcoin instance with `wallet-auth` set to true.
+
+```
```

Comment:
Lets give this a language, like the javascript examples above. `bash` should work. This is the syntax highlighter we're using: http://prismjs.com/index.html#languages-list

Reply:
I was looking for something like ```bash but didn't find it, thanks.